docs/errors/E077.md

```diff
@@ -0,0 +1,15 @@
+# E077: error function call before declaration in blocked scope
+
+A function cannot be called before it's declaration in blocked scope.
```

Comment:
Actually, the standard says that it may or may not be callable. Different implementations behave differently. I think this is worth mentioning. E.g. say that a function can't be called before its declaration in a block scope *in Safari*.